File: Day01/Php/main.php
<?php

function openFile($filename)
{
    $file = fopen($filename, "r");

    if (!$file)
        return "";

    $fileContent = fread($file, filesize($filename));
    fclose($file);

    return $fileContent;
}

function getFirstNumber($input, $wordsToFind)
{
    for ($i = 0; $i < strlen($input); $i++) {
        if (ctype_digit($input[$i]))
            return $input[$i];

        foreach ($wordsToFind as $index => $word)
            if (substr($input, $i, strlen($word)) == $word)
                return $index + 1;
    }

    return -1;
}

function getLastNumber($input, $wordsToFind)
{
    for ($i = strlen($input) - 1; $i >= 0; $i--) {
        if (ctype_digit($input[$i]))
            return $input[$i];

        foreach ($wordsToFind as $index => $word)
            if (substr($input, $i - strlen($word), strlen($word)) == $word)
                return $index + 1;
    }

    return -1;
}

function parseArray($input)
{
    $data = explode("\n", $input);
    $res = 0;
    $wordsToFind = array("one", "two", "three", "four", "five", "six", "seven", "eight", "nine");

    for ($i = 0; $i < count($data); $i++) {
        if ($data[$i] != "") {
            $first = getFirstNumber($data[$i], $wordsToFind);
            $second = getLastNumber($data[$i], $wordsToFind);

            if ($first != -1 && $second != -1)
                $res += intval($first . $second);
        }
    }

    return $res;
}

$input = openFile("data.txt");

if ($input == "") {
    echo "Error: File not found\n";
    return -1;
}

echo parseArray($input) . "\n";

return 0;

File: Day01/Go/main.go
package main

import (
	"fmt"
	"os"
	"bufio"
)

func isDigit(char byte) bool {
	return char >= '0' && char <= '9'
}

func parseLine(line string, words [9]string) [2]int {
	res := [2]int{-1, -1}

	for i, j := 0, len(line)-1; i < len(line) && (res[0] == -1 || res[1] == -1); i, j = i+1, j-1 {
		if isDigit(line[i]) && res[0] == -1 {
			res[0] = int(line[i]) - '0'
		}
		if isDigit(line[j]) && res[1] == -1 {
			res[1] = int(line[j]) - '0'
		}
		for index, word := range words {
			if i+len(word) <= len(line) && line[i:i+len(word)] == word && res[0] == -1 {
				res[0] = index + 1
			}
			if j-len(word)+1 >= 0 && line[j-len(word)+1:j+1] == word && res[1] == -1 {
				res[1] = index + 1
			}
		}
	}
	return res
}

func parseArray(input *bufio.Scanner) int {
	var res int
	words := [9]string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}
	nb := [2]int{-1, -1}
	for input.Scan() {
		var line = input.Text()
		nb = parseLine(line, words)
		if line != "" && nb[0] != -1 && nb[1] != -1 {
			res += ((nb[0] * 10) + nb[1])
		}

	}
	return res
}

func main() {
	fileName := "message.txt"
	file, err := os.Open(fileName)
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	var buf = bufio.NewScanner(file)
	if err := buf.Err(); err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	res := parseArray(buf)
	defer file.Close()
	fmt.Println(res)
}
